maven-resolver-provider/src/main/java/org/apache/maven/repository/internal/DefaultArtifactDescriptorReader.java

```diff
@@ -319,10 +323,20 @@ private Model loadPom( RepositorySystemSession session, ArtifactDescriptorReques
             if ( relocation != null )
             {
                 result.addRelocation( a );
-                a =
+                Artifact relocatedArtifact =
                     new RelocatedArtifact( a, relocation.getGroupId(), relocation.getArtifactId(),
                                            relocation.getVersion() );
-                result.setArtifact( a );
+                if ( LOGGER.isWarnEnabled() )
+                {
+                    String message = "The artifact " + a + " has been relocated to " + relocatedArtifact;
+                    if ( relocation.getMessage() != null )
+                    {
+                        message += ": " + relocation.getMessage();
+                    }
+                    LOGGER.warn( message );
+                }
+                result.setArtifact( relocatedArtifact );
+                a = relocatedArtifact;
```

Comment:
I wonder why this line is necessary....

Reply:
That's the fix for the bug I found previously

Reply:
I know, but why is redefining this variable which isn't used in the loop anymore fixes the issue?

Reply:
I had to read twice to understand that the loop is initialized only once and this variable is used again in the next iteration :)

Reply:
Alright, then this makes sense otherwise it'd be moot.